frontend/src/components/ViewActivities/activitylist.js

```diff
@@ -8,17 +8,33 @@ import app from '../Firebase';
 const db = app.firestore();
 
 /**
- * Gets the list of activities from the server. 
+ * ReactJS class component for the list of activities. 
  * 
- * @param {string} tripId The trip ID.
- * @return {ActivityInfo[]} The list of trip activities.
+ * @property {Object} props ReactJS props.
+ * @property {string} tripId The tripID.
  */
-export async function getActivityList(tripId) {
-    let tripActivities = [];
-    
-    return db.collection(DB.COLLECTION_TRIPS).doc(tripId)
-    .collection(DB.COLLECTION_ACTIVITIES).get()
-    .then(querySnapshot => {
+class ActivityList extends React.Component {
+  /** @override */
+  constructor(props) {
+    super(props);
+    this.state = { days : [] };
+
+    this.getActivityList = this.getActivityList.bind(this);
+  }
+
+  /**
+   * Gets the list of activities from the server. 
+   * 
+   * This function sets `this.state.days` to the sorted days.
+   * 
+   * @param {string} tripId The trip ID.
+   */
+  async getActivityList(tripId) {
+    db.collection(DB.COLLECTION_TRIPS).doc(tripId)
+    .collection(DB.COLLECTION_ACTIVITIES)
+    .onSnapshot(querySnapshot => {
+      let tripActivities = [];
+      this.setState({days: []});
```

Comment:
What does this line do?

Reply:
Clear the current state of the activities. Prevents everything from showing up twice every time there's an edit.